pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -434,7 +434,7 @@ def _get_betas_and_extra_headers(
         if has_strict_tools or model_request_parameters.output_mode == 'native':
             betas.add('structured-outputs-2025-11-13')
 
-        if beta_header := extra_headers.pop('anthropic-beta', None):
+        if beta_header := extra_headers.get('anthropic-beta', None):
             betas.update({stripped_beta for beta in beta_header.split(',') if (stripped_beta := beta.strip())})
 
         return betas, extra_headers
```

Comment:
I believe we pass the whole list of `betas` to the messages creation method later; is that not sufficient?

A test would be great!